src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,371 @@
+"""Jobs for running phonon calculations."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+
+import numpy as np
+from jobflow import Flow, Response, job
+from phonopy import Phonopy
+from phonopy.units import VaspToTHz
+from pymatgen.core import Structure
+from pymatgen.io.phonopy import get_phonopy_structure, get_pmg_structure
+from pymatgen.phonon.bandstructure import PhononBandStructureSymmLine
+from pymatgen.phonon.dos import PhononDos
+from pymatgen.transformations.advanced_transformations import (
+    CubicSupercellTransformation,
+)
+from atomate2.vasp.schemas.phonons import ForceConstants
+from atomate2.common.schemas.math import Matrix3D
+from atomate2.vasp.jobs.base import BaseVaspMaker
+from atomate2.vasp.schemas.phonons import PhononBSDOSDoc
+from atomate2.vasp.sets.base import VaspInputGenerator
+from atomate2.vasp.sets.core import StaticSetGenerator
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "get_total_energy_per_cell",
+    "get_supercell_size",
+    "generate_phonon_displacements",
+    "run_phonon_displacements",
+    "generate_frequencies_eigenvectors",
+    "PhononDisplacementMaker",
+]
+
+
+@job
+def get_total_energy_per_cell(
```

Comment:
Most (if not all) of these functions already exist in: https://github.com/materialsproject/atomate2/blob/main/src/atomate2/common/jobs/phonons.py